llama_stack/core/library_client.py

```diff
@@ -237,8 +253,18 @@ def __init__(
         self.custom_provider_registry = custom_provider_registry
         self.provider_data = provider_data
         self.route_impls: RouteImpls | None = None  # Initialize to None to prevent AttributeError
+        self._is_initialized: bool = False  # Track initialization state
 
     async def initialize(self) -> bool:
+        """
+        Initialize the async client. Can be called multiple times safely.
+
+        Returns:
+            bool: True if initialization was successful
+        """
+        if self._is_initialized:
```

Comment:
how is avoiding double init related to the change?

Reply:
so this flag is to avoid re-init, since we use `auto_init` 

please correct me if i am wrong

Reply:
no this is not right. what you need to do instead is this:

- rename this method to `initialize_impl()` 
- make it synchronous
- you need to mirror the `loop.run_until_complete()` logic here since the method _has_ to be synchronous
- call this method from this constructor only 
- you don't need to do anything inside the LlamaStackClient (synchronous) constructor at all. 

Please take the time to understand the logic and the flow before attempting changes. Do _not_ use AI.

Reply:
@ashwinb `loop.run_until_complete()` does not have any impl 

```
 def run_until_complete(self, future):
        """Run the event loop until a Future is done.

        Return the Future's result, or raise its exception.
        """
        raise NotImplementedError
```

please correct me if i am wrogn

Reply:
@mattf @ashwinb ptal 👍🏽